src/server/service/kv.rs

```diff
@@ -1207,12 +1212,20 @@ fn future_batch_get<E: Engine, L: LockManager>(
     storage: &Storage<E, L>,
     mut req: BatchGetRequest,
 ) -> impl Future<Output = ServerResult<BatchGetResponse>> {
+    let mut statistics = Statistics::default();
     let keys = req.get_keys().iter().map(|x| Key::from_raw(x)).collect();
-    let v = storage.batch_get(req.take_context(), keys, req.get_version().into());
+    let v = storage.batch_get(
+        req.take_context(),
+        keys,
+        req.get_version().into(),
+        &mut statistics);
 
     async move {
+        let perf_statistics = PerfStatisticsInstant::new();
```

Comment:
Before `await` and after `await`, there is no guarantee that the running thread is the same, which may break the thread local `PerfStatisticsInstant`. (note that there is a bug that it is not marked as `!Send`)

Maybe in addition to `&mut statistics`, you also need `&mut perf_statistics`.